src/app/qgsmapsavedialog.cpp

```diff
@@ -92,6 +107,12 @@ QgsMapSaveDialog::QgsMapSaveDialog( QWidget *parent, QgsMapCanvas *mapCanvas, co
 
     this->setWindowTitle( tr( "Save map as PDF" ) );
   }
+
+  connect( buttonBox, &QDialogButtonBox::accepted, this, &QgsMapSaveDialog::accepted );
+  connect( buttonBox, &QDialogButtonBox::rejected, this, [ this ]
+  {
+    delete this;
```

Comment:
Setting [`WA_DeleteOnClose`](http://doc.qt.io/qt-5/qt.html#WidgetAttribute-enum) (from outside) sounds better to me.
The current approach crashes if someone instantiates a `QgsMapSaveDialog` on the stack.

```
dlg->setAttribute( Qt::WA_DeleteOnClose );
```